frontend/src/hooks/api/integrations/queries.tsx

```diff
@@ -110,3 +111,28 @@ export const useDeleteIntegration = () => {
     }
   });
 };
+
+export const useSyncIntegration = (pollingRef: MutableRefObject<NodeJS.Timeout | null>) => {
+  const queryClient = useQueryClient();
+
+  return useMutation<{}, {}, { id: string; workspaceId: string; lastUsed: string }>({
+    mutationFn: ({ id }) => apiRequest.post(`/api/v1/integration/${id}/sync`),
```

Comment:
Its better to remove this logic and use the `refreshInterval` option of react-query for the `get workspace integrations` hook and do a polling for 3.5s automatically. 

The issue with below approach is what if someone else triggers the integration sync in mean time thus changing the last used to something else. Thus the spinner will keep spinning. And i can see more edge cases with this. 

So its easier if we can just add the get integration hook as a polling one so even if anyone. else make changes it will also be seen

Reply:
agree. If we do it this way, what should the user-feedback be upon clicking the Manual sync button? should we just display a notification that says "Successfully triggered manual sync" instead?

Reply:
Yah that would be fine.